packages/core/test/behaviors.test.ts

```diff
@@ -86,7 +86,31 @@ describe('promise behavior (fromPromise)', () => {
     expect(called).toBe(false);
   });
 
-  it('should persist a promise', (done) => {
+  it('should persist an unresolved promise', (done) => {
+    const promiseBehavior = fromPromise(
+      () =>
+        new Promise<number>((res) => {
+          setTimeout(() => res(42), 10);
+        })
+    );
+
+    const actor = interpret(promiseBehavior);
+    actor.start();
+
+    const resolvedPersistedState = actor.getPersistedState();
+    actor.stop();
```

Comment:
The problem here was that `.stop()` mutated referenced contained in the `getPersistedState`, right?

Reply:
Yes